pandas/io/excel.py

```diff
@@ -465,7 +465,7 @@ def _parse_cell(cell_contents, cell_typ):
                 if not squeeze or isinstance(output[asheetname], DataFrame):
                     output[asheetname].columns = output[
                         asheetname].columns.set_names(header_names)
-            except StopIteration:
+            except EmptyDataError:
```

Comment:
this is a USER of the parser. You simply changed the error message. That doesn't solve anything. You need to differentiate them.